src/firefly/js/ui/FileUpload.jsx

```diff
@@ -9,6 +9,40 @@ import LOADING from 'html/images/gxt/loading.gif';
 import {upload} from '../rpc/CoreServices.js';
 
 
+const FileDropZone = ({onChange}) => {
+    const [isDragging, setIsDragging] = useState(false);
+
+    // Prevent the default behavior for drag events to allow dropping files
+    const handleDragEnter = (e) => {
+        e.preventDefault();
+        setIsDragging(true);
+    };
+
+    const handleDragLeave = () => {
+        setIsDragging(false);
+    };
+
+    const handleDrop = (e) => {
```

Comment:
Consider using `useCallback` when passing functions to another component to avoid unnecessary re-renders.